rust/opendp/src/meas/histogram_alp/mod.rs

```diff
@@ -0,0 +1,398 @@
+    where T : num::Float {
+    let mut prefix_sum = Vec::with_capacity(v.len() + 1usize);
+    prefix_sum.push(0);
+
+    v.iter().map(|b| if *b {1} else {-1}).for_each(|x| prefix_sum.push(prefix_sum.last().unwrap() + x));
+    
+    let high = prefix_sum.iter().max().unwrap();
+    let peaks = prefix_sum.iter().enumerate()
+            .filter_map(|(idx, height)| if high == height {Some(idx)} else {None}).collect::<Vec<_>>();
+    
+    // Return the average position
+    T::from(peaks.iter().sum::<usize>()).unwrap() / T::from(peaks.len()).unwrap()
+}
+
+fn compute_estimate<K, T>(state: &AlpState<K, T>, key: &K) -> T
+    where T: num::Float {
+    let v = state.h.iter().map(|f| state.z[f(key) % state.z.len()]).collect::<Vec<_>>();
+
+    estimate_unary::<T>(&v) * T::from(state.alpha).unwrap() / state.scale
+}
+
+pub fn make_alp_histogram<K, C, T>(n: usize, alpha: T, scale: T, s: usize, h: HashFunctions<K>)
+        -> Fallible<Measurement<SizedHistogramDomain<K, C>,
+                                AlpDomain<K, T>,
+                                L1Distance<C>, MaxDivergence<T>>>
+    where K: 'static + Eq + Hash + CheckNull,
+          C: 'static + Clone + Integer + CheckNull + DistanceConstant<C> + InfCast<T> + ToPrimitive,
+          T: 'static + num::Float + DistanceConstant<T> + CastInternalReal + InfCast<C>,
+          AlpState<K,T> : CheckNull {
+    
+    if alpha.is_sign_negative() || alpha.is_zero() {
+        return fallible!(MakeMeasurement, "alpha must be positive")
+    }
+    if scale.is_sign_negative() || scale.is_zero() {
+        return fallible!(MakeMeasurement, "scale must be positive")
+    }
+    if s == 0 {
+        return fallible!(MakeMeasurement, "s can not be zero")
+    }
+    if check_parameters(alpha, scale) {
+        return fallible!(MakeMeasurement, "scale divided by alpha must be above 2^-52")
+    }
+    
+    Ok(Measurement::new(
+        SizedDomain::new(MapDomain { key_domain: AllDomain::new(), value_domain: AllDomain::new() }, n),
+        AllDomain::new(),
+        Function::new_fallible(move |x: &HashMap<K, C>| {
+            let z = compute_projection(x, &h, alpha, scale, s)?;
+            Ok(AlpState { alpha, scale, h:h.clone(), z })
+        }),
+        L1Distance::default(),
+        MaxDivergence::default(),
+        PrivacyRelation::new_from_constant(scale)
+    ))
+}
+
+pub fn make_alp_histogram_parameterized<K, C, T>(n: usize, alpha: T, scale: T, beta: C, size_factor: u32) 
+        -> Fallible<Measurement<SizedHistogramDomain<K, C>, 
+                                AlpDomain<K, T>, 
+                                L1Distance<C>, MaxDivergence<T>>>
+    where K: 'static + Eq + Hash + Clone + CheckNull,
+          C: 'static + Clone + Integer + CheckNull + DistanceConstant<C> + InfCast<T> + ToPrimitive,
+          T: 'static + num::Float + DistanceConstant<T> + CastInternalReal + InfCast<C>,
+          AlpState<K,T> : CheckNull {
+
+    let beta: f64 = T::inf_cast(beta).unwrap().to_f64()
```

Comment:
Minor, but you don't have to unwrap here, you can just try with `?`: `T::inf_cast(beta)?.to_f64()`